drivers/syslog/syslog_filechannel.c

```diff
@@ -58,25 +60,28 @@ FAR static struct syslog_channel_s *g_syslog_file_channel;
  * Private Functions
  ****************************************************************************/
 
-#ifdef CONFIG_SYSLOG_FILE_ROTATE
+#if CONFIG_SYSLOG_FILE_ROTATIONS > 0
 static void log_rotate(FAR const char *log_file)
 {
+  int i;
   int fd;
   off_t size;
   struct stat f_stat;
-  char *backup_file;
+  size_t name_size;
+  FAR char *rotate_to;
+  FAR char *rotate_from;
 
   /* Get the size of the current log file. */
 
-  fd = open(log_file, O_RDONLY);
+  fd = nx_open(log_file, O_RDONLY);
   if (fd < 0)
     {
       return;
     }
 
   fstat(fd, &f_stat);
```

Comment:
could simplify nx_open/fstat/nx_close to stat.

Reply:
Correct, thank you.